android/src/main/java/com/terveystalo/react_native/matomo_sdk/RNMatomoSdkModule.java

```diff
@@ -77,11 +86,45 @@ public void trackEvent(String category, String action, ReadableMap optionalParam
       if (optionalParameters.hasKey("value")) {
         event.value((float) optionalParameters.getDouble("value"));
       }
+
+      insertCustomDimensions(tracker);
+
       event.with(tracker);
 
       promise.resolve(null);
     } catch(Exception e) {
       promise.reject(e);
     }
   }
-}
\ No newline at end of file
+
+  @ReactMethod
+  @SuppressWarnings("unused")
+  public void setCustomDimension(int dimensionId, @Nullable String value, Promise promise) {
+    try {
+      if (value != null) {
+        mCustomDimensions.add(new CustomDimension(dimensionId, value));
```

Comment:
What happens if a value with the given id already exists? Then there are two, possibly different, values in the dimensions. Which one will be sent to the server?